snowfall/tools/net.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c)  2021  Xiaomi Corp.       (author: Fangjun Kuang)
+
+import torch
+import lhotse
+
+
+def compute_post(model: str,
+                 feats: str,
+                 output_dir: str,
+                 device: torch.device,
+                 max_duration: int = 200) -> None:
+    '''Compute posteriors given a model and a FeatureSet.
+
+    Args:
+      model:
+        Filename to the torch scripted model.
+      feats:
+        Filename to the FeatureSet manifest.
+      output_dir:
+        Output directory.
+      device:
+        It indicates which device to use for running the network.
+      max_duration:
+        Maximum duration in seconds in a batch.
+    Returns:
+      Return None.
+    '''
+    scripted_model = torch.jit.load(model)
+    scripted_model.eval()
+
+    scripted_model.to(device)
+
+    cuts = lhotse.load_manifest(feats)
+
+    dataset = lhotse.dataset.K2SpeechRecognitionDataset(cuts, return_cuts=True)
+
+    sampler = lhotse.dataset.SingleCutSampler(cuts, max_duration=max_duration)
+
+    dataloader = torch.utils.data.DataLoader(dataset,
+                                             batch_size=None,
+                                             sampler=sampler,
+                                             num_workers=1)
+
+    output_dir = Path(output_dir)
+    output_dir.mkdir(exist_ok=True)
+    storage_path = output_dir / 'posts'
+
+    posts_writer = lhotse.NumpyFilesWriter(storage_path=storage_path)
```

Comment:
This is going to create a lot of files, not sure if NumpyHdf5Writer is preferable